src/types/adapters/IntegrationAdapter.ts

```diff
@@ -120,13 +113,32 @@ export const toIntegration = (serverIntegration: ServerIntegrationResponse): Int
 };
 
 export const toIntegrations = (serverIntegrations: Array<ServerIntegrationResponse>): Array<Integration> => {
-    return filterOutDefaultAction(serverIntegrations).map(toIntegration);
+    return serverIntegrations
+    .filter(filterOutDefaultAction)
+    .map(toIntegration);
 };
 
 type ServerIntegrationProperties = Schemas.EmailSubscriptionProperties | Schemas.WebhookProperties | Schemas.CamelProperties
 
 export const toIntegrationProperties = (integration: Integration | NewIntegration): ServerIntegrationProperties => {
-    switch (integration.type) {
+
+    const type = integration.type;
+
+    if (isCamelType(type)) {
+        const integrationCamel: IntegrationCamel = integration as IntegrationCamel;
+        return {
+            url: integrationCamel.url,
+            disable_ssl_verification: !integrationCamel.sslVerificationEnabled,
+            secret_token: integrationCamel.secretToken,
```

Comment:
Is a `secret_token` a new parameter for a integration? So far we've used `extras[token]` to pass the token for our [Splunk integration](https://github.com/RedHatInsights/eventing-integrations/blob/efea5f7cd5e253739a19ccf206f52063f0fd4da2/splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java#L151).
cc @pilhuhn

Reply:
I don't think it's new.